src/node_file.cc

```diff
@@ -1994,8 +1997,10 @@ static void CopyFile(const FunctionCallbackInfo<Value>& args) {
   THROW_IF_INSUFFICIENT_PERMISSIONS(
       env, permission::PermissionScope::kFileSystemWrite, dest.ToStringView());
 
-  CHECK(args[2]->IsInt32());
-  const int flags = args[2].As<Int32>()->Value();
+  int flags;
+  if (!GetValidFileMode(env, args[2], UV_FS_COPYFILE).To(&flags)) {
+    return;
+  }
```

Comment:
I believe for consistency, the check should be performed **before** the `THROW_IF_INSUFFICIENT_PERMISSIONS`. Otherwise, the error message will change to users that:

* Don't have access to the file (due to permission model) and
* Don't pass a valid file mode

Reply:
Yes, definitely makes sense. Updating now.